pkg/controller/ambassadorinstallation/scout.go

```diff
@@ -0,0 +1,87 @@
+// From ambassador/internal/pkg/edgectl/scout.go with some modifications & simplifications.
+
+package ambassadorinstallation
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/datawire/ambassador-operator/version"
+	"github.com/datawire/ambassador/pkg/metriton"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+)
+
+// The Scout structure maintains an index, which is the count of calls
+// to Report, incrementing each call.  This provides a sequence of actions
+// to make it easier to search through the reports.
+// The Reporter is simply the Metriton reporting object.
+type Scout struct {
+	index    int
+	Reporter *metriton.Reporter
+}
+
+// Metadata is simply a key and an untyped value, instances passed in as parameters
+// to (s *Scout) Report
+type ScoutMeta struct {
+	Key   string
+	Value interface{}
+}
+
+// Create a new Scout object, with a parameter stating what the Scout instance
+// will be reporting on.  The Ambassador Operator may be installing, updating,
+// or deleting the Ambassador installation.
+func NewScout(mode string, installID types.UID) (s *Scout) {
+	return &Scout{
+		Reporter: &metriton.Reporter{
+			Application: "ambassador-operator",
+			Version:     version.Version,
+			GetInstallID: func(r *metriton.Reporter) (string, error) {
+				return string(installID), nil
+			},
+			// Fixed (growing) metadata passed with every report
+			BaseMetadata: map[string]interface{}{
+				"mode":     mode,
+				"trace_id": uuid.New().String(),
+			},
+		},
+	}
```

Comment:
I would do 
```golang
	return &Scout{
		index: 0,
		Reporter: &metriton.Reporter{
			Application: "ambassador-operator",
			Version:     version.Version,
			GetInstallID: func(r *metriton.Reporter) (string, error) {
				return string(installID), nil
			},
			// Fixed (growing) metadata passed with every report
			BaseMetadata: map[string]interface{}{
				"mode":     mode,
				"trace_id": uuid.New().String(),
			},
		},
	}
```
so caller would not need to invoke `Reset()` (as it does not make much sense to reset something you have just created).

Reply:
good point, I will do that.